internal/pkg/vcsutil/doc.go

```diff
@@ -0,0 +1,4 @@
+// Package vcsutil provides utility functions for inspecting the state of a local git repository.
+//
+// Currently, vcsutil executes the `git` command, so git must be installed for it to work.
+package vcsutil
```

Comment:
Nitpick, I tend to dislike "util" in the package names cause its kind of a kitchen sink (i.e. https://dave.cheney.net/2019/01/08/avoid-package-names-like-base-util-or-common). And given this is git-specific for now and indefinitely, may I suggest `gitdirty`?

Its easy enough to change package names later, especially internal ones.